### PR TITLE
Installation details for windows users in contributing.md and Updated requiremments.txt

### PR DESCRIPTION
**Pull request by tusharrao198**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,53 +1,121 @@
...
+   2. **On Linux** - Activate the virtual environment  
+         `source venv/bin/activate`  
+
+      **On Windows** - Activate the virtual environment  
+         `venv\Scripts\activate`  
+         
+   3. Install required modules  
+      `pip install -r requirements.txt`  
 
 5.  Store your secrets by adding them at the end of `env/bin/activate`. They will be set whenever you run the virtual environment.
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:57]: Change `env` to `venv`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,53 +1,121 @@
...
+
+   2. Create a local postgres database. Keep the credentials handy, you'll need them to link the database to django.  
+    **Steps to create a postgresql db on your system**  
+      **On linux** -  
+       1. `sudo su - postgres`  
+       2. `psql`  
+       3. CREATE DATABASE <db-name>;  
+       4. CREATE USER <db-username> WITH PASSWORD <db-password>;  
+       5. ALTER ROLE <DB-USERNAME> SET client_encoding TO 'utf-8';  
+       6. GRANT ALL PRIVILEGES ON DATABASE <db-name> TO <db-username>;
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:34]: Use backticks for formatting.

Comment (gajrajgchouhan) [reply CONTRIBUTING.md:34]: and enclose `<db-password>` in single quotes.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,53 +1,121 @@
...
+
+        `sudo apt install postgresql postgresql-contrib`  
+
+      **On Windows** -     
+            Install -     
+                  1. PostgreSQL - `https://www.postgresql.org/download/windows/`. 
+   
+                        **( Note :- Remeber Password entered here.)**.  
+                                 
+                  2. pgAdmin - `https://www.pgadmin.org/download/pgadmin-4-windows/`
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:22]: The formatting could be improved, this is due to the newlines and backticks. Try changing them and backticks could be removed from the URLs.

File: CONTRIBUTING.md (hunk 1 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: updated..

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,53 +1,121 @@
...
+    `python manage.py check`    
 
-7.  Run the server on your machine.
+    Run the server on your machine.   
     
-    `python manage.py runserver`
+    `python manage.py runserver`   
     
-    The application will be running at `localhost:8000`.
+    The application will be running at `localhost:8000`.
```
Comment (LalitNM) [inline CONTRIBUTING.md:114]: Remove trailing spaces.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,53 +1,121 @@
-## How to Contribute
+## How to Contribute  
 
-1. Fork this repository to your github account.
+1. Fork this repository to your github account.
```
Comment (LalitNM) [inline CONTRIBUTING.md:3]: Remove trailing spaces.

File: .gitignore [hunk 6]
```diff
@@ -6,10 +6,10 @@ __pycache__/
 
 # C extensions
 *.so
-
 # Distribution / packaging
 .Python
 env/
+.env
```
Comment (LalitNM) [inline .gitignore:12]: remove this as we are not going to use this method of storing variables.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,53 +1,121 @@
...
+                    'HOST': config('DB_HOST'),  
+                    'PORT': '',  
+                }  
+            }  
+
+        7. Since, .env file conatins all of sensitive data. Before uploading, add `.env` file to gitignore to prevent sensitve data uploading on
+        github.  
+            To add, just open `.gitignore`, add `.env`. Save file.   ( By Default, already present.)  
+
+
```
Comment (LalitNM) [inline CONTRIBUTING.md:101]: Remove this we don't want to confuse people, only one way is enough.

File: requirements.txt [hunk 8]
```diff
@@ -1,6 +1,28 @@
...
+python3-openid==3.2.0
 pytz==2020.1
-sqlparse==0.3.1
+requests==2.24.0
+requests-oauthlib==1.3.0
+six==1.15.0
+social-auth-app-django==4.0.0
+social-auth-core==3.3.3
+urllib3==1.25.10
+sqlparse==0.3.1
```
Comment (LalitNM) [inline requirements.txt:28]: Remove unnecessary requirements, the server is well running with existing requirements.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md (hunk 3 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: improved.

**Review by LalitNM — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (tusharrao198) [reply .gitignore:12]: done.

File: CONTRIBUTING.md (hunk 7 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,70 +1,73 @@
...
+                        **( Note :- Remeber Password entered here.)**  
+            2. pgAdmin - `https://www.pgadmin.org/download/pgadmin-4-windows/`   
+                            
+   2. Create a local postgres database. Keep the credentials handy, you'll need them to link the database to django.  
+    **Steps to create a postgresql db on your system**  
+      **On linux** -  
+       1. `sudo su - postgres`  
+       2. `psql`  
+       3. CREATE DATABASE <db-name>;  
+       4. CREATE USER <db-username> WITH PASSWORD `<db-password>`;
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:24]: Replace back-tick with single quotes.

**Review by gajrajgchouhan — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 9 above)
Comment (gajrajgchouhan) [reply CONTRIBUTING.md]: And use backticks for the whole line.

File: CONTRIBUTING.md (hunk 4 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 9 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: Done

File: requirements.txt (hunk 8 above)
Comment (tusharrao198) [reply requirements.txt]: Done

File: CONTRIBUTING.md (hunk 5 above)
Comment (tusharrao198) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,70 +1,71 @@
...
+3. Database Setup  
+   1. You should have postgresql installed on your system. If not, run      
+      **On Linux** -    
+        `sudo apt update`  
+        `sudo apt install postgresql postgresql-contrib`    
+
+      **On Windows** -     
+         Install -     
+            1. PostgreSQL - `https://www.postgresql.org/download/windows/`   
+                        **( Note :- Remeber Password entered here.)**
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:15]: Fix typo.

Comment (gajrajgchouhan) [reply CONTRIBUTING.md:15]: And remove backticks on the URLs.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,70 +1,71 @@
...
+      `pip install -r requirements.txt`  
+
+5.  Store your secrets by adding them at the end of `venv/bin/activate`. They will be set whenever you run the virtual environment.
+
+        ```bash
+        export SECRET_KEY=''
+        export DB_NAME=''
+        export DB_USER=''
+        export DB_PASS=''
+        ```
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:54]: Add newline below backticks.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,70 +1,71 @@
...
+            1. PostgreSQL - `https://www.postgresql.org/download/windows/`   
+                        **( Note :- Remeber Password entered here.)**  
+            2. pgAdmin - `https://www.pgadmin.org/download/pgadmin-4-windows/`   
+                            
+   2. Create a local postgres database. Keep the credentials handy, you'll need them to link the database to django.  
+    **Steps to create a postgresql db on your system**  
+      **On linux** -  
+       1. `sudo su - postgres`  
+       2. `psql`  
+       3. `CREATE DATABASE <db-name>`;
```
Comment (gajrajgchouhan) [inline CONTRIBUTING.md:23]: Add semicolon inside the backticks.

**Review by LalitNM — APPROVED** [review]

**Comment by LalitNM** [comment]
@tusharrao198 Merged :rocket:

**Pull request merged**